src/main/datalog_console/components/entity.cljs

```diff
@@ -48,32 +50,42 @@
     (entity? col) (str (select-keys col [:db/id]))
     :else (str col)))
 
-(defn entity [conn]
+
+(def rerender (atom 0))  ; This is a hack and ideally we find a way to eliminate the use of this
+
+(defn entity []
   (let [lookup (r/atom "")
-        entity (r/atom nil)]
-    (fn []
-      [:div {:class "w-full h-full overflow-auto pb-5"}
-       [:form {:class "flex items-end"
-               :on-submit
-               (fn [e]
-                 (.preventDefault e)
-                 (reset! entity (d/entity @conn (cljs.reader/read-string @lookup))))}
-        [:label {:class "block pt-1 pl-1"}
-         [:p {:class "font-bold"} "Entity lookup"]
-         [:input {:type "text"
-                  :placeholder "id or [:uniq-attr1 \"v1\" ...]"
-                  :class "border py-1 px-2 rounded w-56"
-                  :value @lookup
-                  :on-change #(reset! lookup (.-value (.-target %)))
-                  }]]
-        [:button {:type "submit" 
-                  :class "ml-1 py-1 px-2 rounded bg-gray-200 border"} 
-         "Get entity"]]
-       (when @entity
-         [c.tree-table/tree-table
-          {:caption (str "entity " (select-keys @entity [:db/id]))
-           :head-row ["Attribute", "Value"]
-           :rows (entity->rows @entity)
-           :expandable-row? expandable-row?
-           :expand-row expand-row
-           :render-col render-col}])])))
+        view-state (r/atom #{})]
+    (fn [conn]
+      (let [entity (d/entity @conn (cljs.reader/read-string @lookup))
+            _ (swap! rerender inc)]
+        [:div {:class "w-full h-full overflow-auto pb-5 stack-small"}
+         [:form {:class "flex items-end"
+                 :on-submit
+                 (fn [e]
+                   (.preventDefault e)
+                   (let [current-lookup (goog.object/getValueByKeys e #js ["target" "elements" "lookup" "value"])]
+                     (when-not (= @lookup current-lookup) (reset! view-state #{}))
+                     (reset! lookup current-lookup)))}
+          [:label {:class "block pl-1"}
+           [:p {:class "font-bold"} "Entity lookup"]
+           [:input {:type "text"
+                    :name "lookup"
+                    :placeholder "id or [:uniq-attr1 \"v1\" ...]"
+                    :class "border py-1 px-2 rounded w-56"}]]
+          [:button {:type "submit"
+                    :class "ml-1 py-1 px-2 rounded bg-gray-200 border shadow-hard btn-border"}
+           "Get entity"]]
+         (when entity
+           ^{:key @rerender}
```

Comment:
I think swapping `@rerender` out for `(str (vals entity))` might do the trick

Reply:
This is only the vals of an `entity`. The problem we have is the `reverse-ref`. What we need I think is a top level `atom` which contains the currently observed `entities` and a `r/atom` which contains the `entity` of that `reverse-ref`. This would trigger those sub components to re-render by just referencing that `r/atom`. This `r/atom` could also be updated with a `go-loop` reading off a channel for when we make it async.